### ISSUE TITLE
Snap just updated to 107.0 (from 106.3) and all DHCP and DNS has stopped working!

### ISSUE THREAD

**Issue by Tloram** [AdguardTeam/AdGuardHome#3976]
Have a question or an idea? Please search it [on our forum](https://github.com/AdguardTeam/AdGuardHome/discussions) to make sure it was not yet asked. If you cannot find what you had in mind, please [submit it here](https://github.com/AdguardTeam/AdGuardHome/discussions/new).

### Prerequisites

Please answer the following questions for yourself before submitting an issue. **YOU MAY DELETE THE PREREQUISITES SECTION.**

- [x] I am running the latest version
- [x] I checked the documentation and found no answer
- [x] I checked to make sure that this issue has not already been filed

### Issue Details

* **Version of AdGuard Home server:**
  * 107.0
* **How did you install AdGuard Home:**
  * Snap on Ubuntu

### Expected Behavior
DHCP and DNS to continue working as per 106.3 did.

### Actual Behavior
Upon the snap auto updating itself to 107.0, all DHCP and DNS appears to have stopped working.

### Additional Information
Have temporarily reverted the snap to previous version (106.3) and immediately everything started working again, network wide.

**Comment by ainar-g** [comment]
@Chinaski1, please try and reproduce this with our Snapcraft setup.

**Comment by gkavv** [comment]
Actually I was about to open a similar issue but only for the DHCP service.
In my case the DNS works without problem, but the DHCP is off.
And the strange thing here is, that I cannot see any error message at logs (i.e. failed to start service or port 67 occupied etc)

**Comment by ilarrain** [comment]
I'm having the same issue.

This are the logs while starting the service ('sudo snap start adguard-home'):
```
$ sudo snap logs adguard-home -f
2021-12-21T23:14:04-03:00 systemd[1]: Started Service for snap application adguard-home.adguard-home.
2021-12-21T23:14:04-03:00 adguard-home.adguard-home[165385]: 2021/12/21 23:14:04.978007 [info] AdGuard Home, version v0.107.0
2021-12-21T23:14:04-03:00 adguard-home.adguard-home[165385]: 2021/12/21 23:14:04.989062 [info] dhcp: loaded leases v4:58  v6:0  total-read:58 from DB
2021-12-21T23:14:05-03:00 adguard-home.adguard-home[165385]: 2021/12/21 23:14:05.008327 [info] Initializing auth module: /var/snap/adguard-home/3353/data/sessions.db
2021-12-21T23:14:05-03:00 adguard-home.adguard-home[165385]: 2021/12/21 23:14:05.008663 [info] auth: initialized.  users:1  sessions:2
2021-12-21T23:14:05-03:00 adguard-home.adguard-home[165385]: 2021/12/21 23:14:05.008736 [info] Initialize web module
2021-12-21T23:14:05-03:00 adguard-home.adguard-home[165385]: 2021/12/21 23:14:05.012948 [info] warning: cannot initialize ipset: ipset: dialing netfilter: dialing v4: socket: invalid argument
2021-12-21T23:14:05-03:00 adguard-home.adguard-home[165385]: 2021/12/21 23:14:05.138155 [error] starting dhcp server: dhcpv4: creating raw udp connection: permission denied
2021-12-21T23:14:05-03:00 adguard-home.adguard-home[165385]: 2021/12/21 23:14:05.139009 [info] AdGuard Home is available at the following addresses:
2021-12-21T23:14:05-03:00 adguard-home.adguard-home[165385]: 2021/12/21 23:14:05.157533 [info] Go to http://127.0.0.1:3000
2021-12-21T23:14:05-03:00 adguard-home.adguard-home[165385]: 2021/12/21 23:14:05.157634 [info] Go to http://[::1]:3000
2021-12-21T23:14:05-03:00 adguard-home.adguard-home[165385]: 2021/12/21 23:14:05.157662 [info] Go to http://192.168.1.1:3000
2021-12-21T23:14:05-03:00 adguard-home.adguard-home[165385]: 2021/12/21 23:14:05.157703 [info] Go to http://172.30.32.1:3000
2021-12-21T23:14:05-03:00 adguard-home.adguard-home[165385]: 2021/12/21 23:14:05.157763 [info] Go to http://172.17.0.1:3000
2021-12-21T23:14:06-03:00 adguard-home.adguard-home[165385]: 2021/12/21 23:14:06.200194 [info] Starting the DNS proxy server
2021-12-21T23:14:06-03:00 adguard-home.adguard-home[165385]: 2021/12/21 23:14:06.200274 [info] Ratelimit is enabled and set to 20 rps
2021-12-21T23:14:06-03:00 adguard-home.adguard-home[165385]: 2021/12/21 23:14:06.200294 [info] The server is configured to refuse ANY requests
2021-12-21T23:14:06-03:00 adguard-home.adguard-home[165385]: 2021/12/21 23:14:06.200315 [info] DNS cache is enabled
2021-12-21T23:14:06-03:00 adguard-home.adguard-home[165385]: 2021/12/21 23:14:06.200343 [info] MaxGoroutines is set to 300
2021-12-21T23:14:06-03:00 adguard-home.adguard-home[165385]: 2021/12/21 23:14:06.200379 [info] Creating the UDP server socket
2021-12-21T23:14:06-03:00 adguard-home.adguard-home[165385]: 2021/12/21 23:14:06.200611 [info] Listening to udp://[::]:53
2021-12-21T23:14:06-03:00 adguard-home.adguard-home[165385]: 2021/12/21 23:14:06.200639 [info] Creating a TCP server socket
2021-12-21T23:14:06-03:00 adguard-home.adguard-home[165385]: 2021/12/21 23:14:06.200761 [info] Listening to tcp://[::]:53
2021-12-21T23:14:06-03:00 adguard-home.adguard-home[165385]: 2021/12/21 23:14:06.200962 [info] Entering the tcp listener loop on [::]:53
2021-12-21T23:14:06-03:00 adguard-home.adguard-home[165385]: 2021/12/21 23:14:06.201005 [info] Entering the UDP listener loop on [::]:53
```

DHCP is not working at all, while DNS is working fine, it was working flawlessly until the snap auto upgrade.

**Comment by yatagasaru** [comment]
I also have the same problem but only for dhcp. DNS is working properly.
```
2021-12-22T03:16:03Z adguard-home.adguard-home[7032]: 2021/12/22 03:16:03.149747 [info] dhcp: loaded leases v4:9  v6:0  total-read:9 from DB
2021-12-22T03:16:14Z adguard-home.adguard-home[7032]: 2021/12/22 03:16:14.280516 [info] dhcp: loaded leases v4:9  v6:0  total-read:9 from DB
2021-12-22T03:16:14Z adguard-home.adguard-home[7032]: 2021/12/22 03:16:14.280617 [info] can't check for static ip; assuming machine has static ip and going on
2021-12-22T03:16:14Z adguard-home.adguard-home[7032]: 2021/12/22 03:16:14.280970 [error] POST /control/dhcp/set_config: enabling dhcp: starting dhcp server: dhcpv4: creating raw udp connection: permission denied
2021-12-22T03:22:24Z adguard-home.adguard-home[7032]: 2021/12/22 03:22:24.991191 [info] dhcp: loaded leases v4:9  v6:0  total-read:9 from DB
2021-12-22T03:22:33Z adguard-home.adguard-home[7032]: 2021/12/22 03:22:33.601257 [info] dhcp: loaded leases v4:9  v6:0  total-read:9 from DB
2021-12-22T03:22:33Z adguard-home.adguard-home[7032]: 2021/12/22 03:22:33.601390 [info] can't check for static ip; assuming machine has static ip and going on
2021-12-22T03:22:33Z adguard-home.adguard-home[7032]: 2021/12/22 03:22:33.601883 [error] POST /control/dhcp/set_config: enabling dhcp: starting dhcp server: dhcpv4: creating raw udp connection: permission denied
```

When I press Enable DHCP Server:
![image](https://user-images.githubusercontent.com/37947184/147031012-c77370e4-b13a-43f8-9c21-72e83f8ae2fe.png)
and then dhcp seems enabled when I reloaded the dashboard but not working.

Check for DHCP Server:
![image](https://user-images.githubusercontent.com/37947184/147031331-d9f2c5ee-aae3-4524-804f-c2931c0e1643.png)

**Comment by rusxakep** [comment]
same problem. reverted to v0.106.3

**Comment by ghokun** [comment]
I can't revert to old version. Is there a workaround ?

**Comment by rusxakep** [comment]
No, only disable DHCP or run adguard in standalone mode (0.106.3 or 0.107).
I was copy all preferences to /opt/, install standalone adguard to /opt and run adguard as service.
Waiting for fix now :(

**Comment by ainar-g** [comment]
@Tloram, @gkavv, @ilarrain, thanks for the reports and the data.  We have confirmed and reproduced the bug.  It seems like the current permissions that our snap is getting are too restrictive.  Could any of you update to the `edge` channel and connect the `network control` plug?

```sh
sudo snap refresh --amend --channel edge adguard-home
sudo snap connect adguard-home:network-control
sudo snap restart adguard-home
```

(You may need to disable and enable the DHCP server in the UI.)  In our tests that fixes the issue.  If we get a confirmation that that fixes it for other people too, we'll request this plug to get connected automatically.

**Comment by gkavv** [comment]
Hello @ainar-g , I just tried the workaround you proposed.
I ran the exact same commands, restarted the service, deactivated and reactivated the dhcp, but still not working :(

**Comment by ghokun** [comment]
> @Tloram, @gkavv, @ilarrain, thanks for the reports and the data. We have confirmed and reproduced the bug. It seems like the current permissions that our snap is getting are too restrictive. Could any of you update to the `edge` channel and connect the `network control` plug?
> 
> ```shell
> sudo snap refresh --amend --channel edge adguard-home
> sudo snap connect adguard-home:network-control
> sudo snap restart adguard-home
> ```
> 
> (You may need to disable and enable the DHCP server in the UI.) In our tests that fixes the issue. If we get a confirmation that that fixes it for other people too, we'll request this plug to get connected automatically.

This did not work for me. Had to install with binary. Can not test it more because I have clients connected to this DHCP.

**Comment by rusxakep** [comment]
Working fine for me:

![image](https://user-images.githubusercontent.com/1502458/147112295-6ae281e1-f097-423f-b3c1-29eaa24b0629.png)

**Comment by ainar-g** [comment]
@rusxakep, thanks!  We've sent a request to the Snap Store to make that capability enabled by default, but I don't know if/when they'll approve it.

@gkavv, @ghokun, I see, thanks for testing.  If any of you (or other people who use the snap and whom the `network-control` solution didn't) help could [enable verbose logging][1] and either attach the (possibly anonymised) logs of the AGH startup here as a file or send them to devteam@adguard.com, that would really help us.

[1]: https://github.com/AdguardTeam/AdGuardHome/wiki/FAQ#verboselog

**Comment by gkavv** [comment]
> @rusxakep, thanks! We've sent a request to the Snap Store to make that capability enabled by default, but I don't know if/when they'll approve it.
> 
> @gkavv, @ghokun, I see, thanks for testing. If any of you (or other people who use the snap and whom the `network-control` solution didn't) help could [enable verbose logging](https://github.com/AdguardTeam/AdGuardHome/wiki/FAQ#verboselog) and either attach the (possibly anonymised) logs of the AGH startup here as a file or send them to [devteam@adguard.com](mailto:devteam@adguard.com), that would really help us.

I just sent my log file at the corresponding email!

**Comment by ainar-g** [comment]
@gkavv, thanks!  We have received it, but we don't see anything wrong with the logs provided.  In fact, we don't see the DHCP server starting at all.  Could you provide logs that appear when you try to check/enable the DHCP server from the UI?  (Having done the `snap connect adguard-home:network-control`.)

**Comment by gkavv** [comment]
> @gkavv, thanks! We have received it, but we don't see anything wrong with the logs provided. In fact, we don't see the DHCP server starting at all. Could you provide logs that appear when you try to check/enable the DHCP server from the UI? (Having done the `snap connect adguard-home:network-control`.)

Done!

**Comment by ainar-g** [comment]
Snapcraft still hasn't replied to our request, so we'll continue monitoring this in v0.107.2.  Until then, the `network-control` solution should fix it in some cases.

@gkavv, thanks again!  Unfortunately, we still don't see anything wrong in the logs.  Can you please file a new issue about your situation, including the last version that worked, error messages in the UI, and why you couldn't enable DHCP from the UI?  Thanks!

**Comment by gkavv** [comment]
@ainar-g OK I will open an issue about the problem that I cannot enable DHCP through frontend.
But still if I enable it directly from the yaml file it shows the server enabled in UI, but it does not actually work.
So, the serious thing here is not the UI problem, but the DHCP not working.
Last time worked on 106.3, as mentioned in this issue.

Anyway thank you!
I will try some things and if it does not work again, I'll most probably switch to another solution.

**Comment by gkavv** [comment]
@ainar-g Hi again,
I tried something else and it seems it worked.
First of all, I wasn't able to enable the DHCP from UI because I haven't before clicked the "check for DHCP servers". After I clicked there, I was able to enable the DHCP from UI.
That one, also solved the other issue (together with the fix you proposed initially here: https://github.com/AdguardTeam/AdGuardHome/issues/3976#issuecomment-999594578)
So, I assume that the "check for DHCP servers" button is also triggering something else in order to unblock DHCP function.

So, everything good from my side too.

**Comment by Tloram** [comment]
@ainar-g I didn't get around to trying your suggestions yet... But in the meanwhile I noticed the snap auto updated to 107.2 and seems to be working fine now. I didn't do anything to fix the issue, it just seems to work ok with this new version.

**Comment by ainar-g** [comment]
Snap Store has granted us the required permissions, so the issues caused by the permission mismatch should be solved either by now or by the time v0.107.3 is released.  Thanks again everyone for testing!
